### PR TITLE
Cryptanalysis of data after a ransomware attack

### LINKED ISSUE #2996: Cryptanalysis of Data after A Ransomware Attack
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Cryptanalysis of Data after A Ransomware Attack

### Introduction paragraph (2-3 paragraphs):
This article seeks to explore the concept of cryptanalysis in light with data encrypted by a malicious actor seeking ransom to decrypt the data. The process of cryptanalysis will be clearly outlined, higlighting the various ethical ways to do the same putting into consideration the possible cryptographic algorithm that may have been employed in encrypting the data.
The possible decryption algorithms will also be identified to be able to help the reader know their possible go to option when they fall prey to a ransomware attack.

### Key takeaways: 
The learner should master the following by the end of studying this article:
1. What is cryptanalysis.
2. Tools used in Cryptanalysis.
3. Methods of encryption.
4.  Methods of decryption.

### References: 
https://www.section.io/engineering-education/cryptography-in-computer-networking/

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by RuthAMare**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2996

### CODE REVIEW

**Comment by Sajeyks** [comment]
Hello @RuthAMare. I shall be reviewing your pr 
cc @hectorkambow

**Comment by RuthAMare** [comment]
Hey @Sajeyks
Okay

File: content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    alt: Cryptanalysis cover image 
+---
+Cryptanalysis is a technique that has been majorly applied to helping find weaknesses in cryptographic algorithms and improve on them to make encryption robust to threats. Cryptanalysis can, however, be used to decipher data encypted by malicous actors to save on the cost of paying the ransom and recover confidential data.
+<!--more-->
+The concepts described in this article will help understand this process.
+
+### Overview
+This article will cover:
+- [Prerequisites](#prerequisites)
+- [What is cryptanalysis?](#what-is-cryptanalysis?)
```
Comment (Sajeyks) [inline content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md:26]: Please remove the question mark in your link description as it is breaking the link.

**Review by Sajeyks — COMMENTED** [review]
Everything looks good, you just have to make a few changes and your article will be ready.

**Comment by Sajeyks** [comment]
this fixes #2996

**Comment by Ericgacoki** [comment]
> In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states "This fixes #(enter your Idea Content Form #)" to link and close your Idea Content (issue) when the PR is merged.
> 
> To help you do it the right way, take a [look](https://github.com/section-engineering-education/engineering-education/pull/3161#issue-703326896) at how other contributors have added theirs.

@Sajeyks, I linked the issue when I came across the first version of this comment.

**Comment by RuthAMare** [comment]
hey @Sajeyks 
Changes requested have been made.

**Comment by Sajeyks** [comment]
@RuthAMare it also seems like your PR contains two different articles and hero images respectively. I suggest you get rid of the other article that seems to have been published already.

File: content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md [hunk 2]
```diff
@@ -16,15 +16,15 @@ images:
   - url: /engineering-education/cryptanalysis-of-data-after-a-ransomware-attack/hero.jpg
     alt: Cryptanalysis cover image 
 ---
-Cryptanalysis is a technique that has been majorly applied to helping find weaknesses in cryptographic algorithms and improve on them to make encryption robust to threats. Cryptanalysis can, however, be used to decipher data encypted by malicous actors to save on the cost of paying the ransom and recover confidential data.
+Cryptanalysis is a technique that has been majorly applied to helping find weaknesses in cryptographic algorithms and improve on them to make encryption robust to threats. Cryptanalysis can, however, be used to decipher data encrypted by malicous actors to save on the cost of paying the ransom and recover confidential data.
```
Comment (Sajeyks) [inline content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md:19]: U forgot to change malicious

**Comment by RuthAMare** [comment]
@Sajeyks Spelling has been corrected and the previous PR deleted

**Comment by Sajeyks** [comment]
> @Sajeyks Spelling has been corrected and the previous PR deleted

Seems like You forgot to remove the 3 images that belong to the previous article as well.

**Comment by RuthAMare** [comment]
@Sajeyks seems it's not possible to delete the images on my end

**Comment by RuthAMare** [comment]
It worked, I guess.
It's done @Sajeyks

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Sajeyks** [comment]
> It worked, I guess.
> It's done @Sajeyks

Great job @RuthAMare

**Comment by srishilesh** [comment]
Moving for the final review as requested by the test peer reviewer.

Cc. @Sajeyks @RuthAMare

File: content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+status: publish
+published:
+url: /cryptanalysis-of-data-after-a-ransomware-attack/
+title: Cryptanalysis of Data after A Ransomware Attack
+description: This article seeks to explore the concept of cryptanalysis in light of data encrypted by a malicious actor seeking ransom to decrypt the data. The process of cryptanalysis will be clearly outlined, highlighting the various ethical ways to do the same putting into consideration the possible cryptographic algorithm that may have been employed in encrypting the data.
+
+The possible decryption algorithms will also be identified to be able to help the reader know their possible go-to option when they fall prey to a ransomware attack.
+author: ruth-mare
+date: 2021-08-06T00:00:00-18:00
+topics: []
```
Comment (srishilesh) [inline content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md:12]:
```suggestion
topics: [Security]
```

File: content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
+---
+layout: engineering-education
+status: publish
+published:
+url: /cryptanalysis-of-data-after-a-ransomware-attack/
+title: Cryptanalysis of Data after A Ransomware Attack
+description: This article seeks to explore the concept of cryptanalysis in light of data encrypted by a malicious actor seeking ransom to decrypt the data. The process of cryptanalysis will be clearly outlined, highlighting the various ethical ways to do the same putting into consideration the possible cryptographic algorithm that may have been employed in encrypting the data.
+
+The possible decryption algorithms will also be identified to be able to help the reader know their possible go-to option when they fall prey to a ransomware attack.
```
Comment (srishilesh) [inline content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md:9]:
```suggestion
description: This article seeks to explore the concept of cryptanalysis in light of data encrypted by a malicious actor seeking ransom to decrypt the data. The process of cryptanalysis will be clearly outlined, highlighting the various ethical ways to do the same putting into consideration the possible cryptographic algorithm that may have been employed in encrypting the data.
```

File: content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+Happy learning!
+
+### Relevant resources
+-[Cryptography in computer](https://www.section.io/engineering-education/cryptography-in-computer-networking/)
+
+-[Understanding AES and RSA encryption Algoritms](https://www.section.io/engineering-education/aes-rsa-encryption/)
+
+-[Cryptanalysis tools](https://resources.infosecinstitute.com/topic/cryptanalysis-tools/)
+
+---
```
Comment (srishilesh) [inline content/articles/cryptanalysis-of-data-after-a-ransomware-attack/index.md:110]:
```suggestion
---
Peer Review Contributions by: [Samuel Mwangi](/engineering-education/authors/samuel-mwangi/)
```

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @RuthAMare 

Could you please commit these changes as well?

@Sajeyks Please make note these suggestions.

**Comment by RuthAMare** [comment]
Hey @srishilesh 
Changes commited.

**Comment by ahmadmardeni1** [comment]
Great work ladies and gentlemen @RuthAMare @Sajeyks @srishilesh 

Please go through the minor changes I made.

**Pull request merged**